internal/error.go

```diff
@@ -523,6 +548,8 @@ func (e *ApplicationError) Unwrap() error {
 	return e.cause
 }
 
+func (e *ApplicationError) NextRetryDelay() time.Duration { return e.nextRetryDelay }
```

Comment:
Should have godoc, but not that big of a deal.

Reply:
I am failing to find documentation which will explain NextRetryDelay better than the name of the method.

Reply:
Godoc rules often stipulate Godoc for all exported items regardless of obviousness (yes that means usually if redundant). Granted we don't have such a strict requirement, but we usually follow it.